src/Symfony/Component/Security/Core/Security.php

```diff
@@ -57,4 +67,58 @@ public function getToken(): ?TokenInterface
     {
         return $this->container->get('security.token_storage')->getToken();
     }
+
+    public function login(UserInterface $user, string $authenticatorName = null, string $firewallName = null): void
```

Comment:
The authenticatorName would be the “key” from the firewall - like form_login? What about custom_authenticators? Are they included in this list? And would the user pass the service id to the custom authenticator?

Reply:
Thanks for your response. The key is the firewall name as defined in the security.yaml. The firewalls are build in the SecurityExtension with the `createFirewall` method. I'm not sure if this takes account of custom authenticators. I would say no.

Reply:
It just missed a few bits for custom authenticators to be handled properly! It's ok now (and tested)
```php
$security->login($user, ApiKeyAuthenticator::class);
```